### PR TITLE
Sketch new feature models; use Row for Assignment message

### PR DESCRIPTION
**Pull request by fritzo**
- Use `Row` message for both values and assignments
- Sketch LatentTree message
- Rename variables towards more general models

Most of the interesting material can be seen in https://github.com/priorknowledge/loom/blob/future-models/src/schema.proto

### CODE REVIEW

File: src/schema.proto [hunk 1]
```diff
@@ -1,38 +1,47 @@
...
+    message CountList {
+      repeated uint32 items = 1;
+    }
+    message RealList {
+      repeated float items = 1;
+    }
+
+    repeated bool observed = 1;
+    repeated bool booleans = 2;  // including bb
+    repeated uint32 counts = 3;  // including dd, pyd, dpd, gp
```
Comment (jglidden) [inline src/schema.proto]: s/pyd/pypd

File: src/schema.proto [hunk 2]
```diff
@@ -44,7 +53,6 @@ message ProductModel {
 }
 
 //----------------------------------------------------------------------------
-// CrossCat
 
 message CrossCat {
   message Kind {
```
Comment (jglidden) [inline src/schema.proto:58]: The asymmetry between kinds and categories is a bit striking. Why are category assignments represented as latent state attached to rows, while kind assignments are represented as objects which possess features?

**Comment by jglidden** [comment]
+1

File: src/schema.proto (hunk 2 above)
Comment (fritzo) [inline src/schema.proto:58]: Good observation. My archaeological view is that messages seem to be nested in the same way the loops are nested in the code that operates on them:

```
for row in rows:
   for kind in kinds:
        for feature in kind:
            ...compute score...
        category_index = sample(...)
        for feature in kind:
            ...update suffstats...
```

**Pull request merged**